server/engine/src/main/java/org/eclipse/lsp/cobol/service/delegates/communications/ServerCommunications.java

```diff
@@ -138,6 +144,60 @@ public void cancelProgressNotification(String uri) {
     uriInProgress.remove(files.decodeURI(uri));
   }
 
+  @Override
```

Comment:
All public methods that are working with `uriInProgress` must be synchronous

Reply:
updated